Engine/Storage/LocalObjectStore.cs

```diff
@@ -250,7 +265,7 @@ public bool Delete(string key)
 
                 try
                 {
-                    var path = GetFilePathForKey(key);
+                    var path = PathForKey(key);
                     if (File.Exists(path))
                     {
                         File.Delete(path);
```

Comment:
I think `PersistData()` should clear the directory before writing to disk and we can remove this delete from here?